### PR TITLE
 ✨ Fixed #JENKINS-27013. add support for managing credentials

### PR DESCRIPTION
**Pull request by wei-lee**
This PR will add support for:

* listing credentials
* create/update/delete credentials

At the moment, it only supports "Username with password" credential type. I am planning to add support for ssh key credential as well. But want to get some feedback first in the mean time.

Thanks.

### CODE REVIEW

**Comment by khmarbaise** [comment]
What kind of feedback did you expect ?

**Comment by wei-lee** [comment]
@khmarbaise anything that you think isn't good, or not following the practices of the library. Like code review basically.

**Comment by wei-lee** [comment]
@khmarbaise I am considering this PR is completed now. Feel free to review the PR. Could appreciate if we can get a SNAPSHOT release of the module after this PR is merged. Thanks a lot.

**Comment by wei-lee** [comment]
@matzew could you publish a new snapshot from this PR to maven central?

**Comment by matzew** [comment]
@wei-lee I can look at publishing a snapshot of this to our JBoss/RedHat repo.

**Comment by wei-lee** [comment]
@khmarbaise we are looking at using this module in one of our open source projects [digger-java](https://github.com/aerogear/digger-java). But we are a bit blocked because there is no release of this PR.

We would like to help in anyway we can. If you are ok with it, I can help to publish a snapshot release of this module. Another option is we can fork this and publish it to our own repo with a different group id.

WDYT?

**Comment by matzew** [comment]
I've deployed a fork to our nexus 

@khmarbaise if we can help, e.g. w/ releases etc, let us know :-)

**Comment by khmarbaise** [comment]
If you could give me some time until tomorrow I have some time to make a SNAPSHOT deploy....

**Comment by wei-lee** [comment]
@khmarbaise no problem at all. Thanks for the help!

**Comment by khmarbaise** [comment]
Hm..one problem is that the reference to JENKINS-27013 is wrong, cause that references a plugin of Jenkins but not this API ?

**Comment by wei-lee** [comment]
@khmarbaise that's really stupid of me 😢 , not sure how I got the wrong issue number.

The right jira is this: https://issues.jenkins-ci.org/browse/JENKINS-44586. I think we can mark that one as resolved?

Thanks.

**Comment by khmarbaise** [comment]
Can you please a new commit containing the pull request with the correct issue reference otherwise the history is hard to follow and confusing...
